aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java

```diff
@@ -417,7 +432,20 @@ private static void registerResource(String s3Location) {
               .build());
       // when a resource is registered, LF will update SLR with necessary permissions which has a
       // propagation delay
-      waitForIamConsistency();
+      Awaitility.await()
+          .pollDelay(Duration.ofSeconds(1))
+          .atMost(Duration.ofSeconds(10))
+          .ignoreExceptions()
+          .untilAsserted(
+              () ->
+                  Assertions.assertThat(
+                          lakeformation
+                              .describeResource(
+                                  DescribeResourceRequest.builder().resourceArn(arn).build())
+                              .resourceInfo()
+                              .roleArn()
+                              .equalsIgnoreCase(lfRegisterPathRoleArn))
```

Comment:
this should do `.isEqualToIgnoringCase(lfRegisterPathRoleArn))` rather than `isTrue()`